File: #228_2015-08-17 [Easy]/alpha.c
#include<stdio.h>
#include<string.h>

#define TRUE  1
#define FALSE 0

void main(int argc, char *argv[]) {
    
    if (argc < 2) {
    	printf("Error: expected one argument\n");
    	return;
    }

int dir = 0, *i, len = sizeof(argv[1]);
   	char *res_str;

    for (i = 0; i < len; ++i) {
    	int *prev = *argv[1]+i;
    	int *curr = *argv[1]+i+1;

    	printf("%n, %n", prev, curr);

    	if( dir == 0 ) {
    		if( prev < curr )
    			dir = 1;
    		else if( prev > curr )
    			dir = -1;
    	}

    	if( dir != 0 ) {
    		if( dir ) {
    			if( prev > curr ) {
    				res_str = "NOT IN ORDER";
    				break;
    			}
    				
    		} else {
    			if( prev < curr ) {
    				res_str = "NOT IN ORDER";
    				break;
    			}
    		}
    	}
    }

    if( res_str != "NOT IN ORDER" ){
    	if( dir ) {
        	res_str = "IN ORDER";
        } else {
        	res_str = "REVERSE ORDER";
        }
    }

    printf( "%s\n", res_str );

}

File: #228_2015-08-17 [Easy]/alpha.js
if(process.argv.length < 3){
	console.log('Expected one argument');
} else {
	var word = process.argv[2];
	
	var prev = word.charCodeAt(0);
	var curr = word.charCodeAt(1);

	var direction = prev < curr ? "forward" : "backward";

	prev = curr;

	for(var i=2; i<word.length; i++) {
		var curr = word.charCodeAt(i);

		var thisDir = prev < curr ? "forward" : "backward";

		if(thisDir != direction) {
			direction = "oh noes!";
			break;
		}
	}

	var message = (direction == "forward" || direction == "oh noes!") ? "IN ORDER" : "REVERSE ORDER"

	if(direction == "oh noes!") {
		message = "NOT ".concat(message);
	}

	console.log(message);
}